### ISSUE TITLE
Common-envelope event: consistent calculation of final separation

### ISSUE THREAD

**Issue by avigna** [TeamCOMPAS/COMPAS#267]
**Describe the bug**
In the ```BaseBinaryStar::ResolveCommonEnvelopeEvent``` function, the lambda structure parameter is calculated using the radius of the stars, but the final separation is calculated using the Roche lobe of the stars. This is wrong and particularly harmful when the radius is not simillar to the Roche-lobe radius. 

**Label the issue**
`urgency_high`     - This is a very urgent issue and should be resolved as soon as possible
`severity_moderate` - This is a moderately severe bug

**To Reproduce**
Run any system with a CEE, easier to notice with a double-core CEE (will add an example later).

**Expected behavior**
1. All the ```BaseBinaryStar::ResolveCommonEnvelopeEvent``` function should only read with Rsol units, instead of a mixture between AU and Rsol.
~2. User should be allowed to choose if they want to use the stellar radius or Roche lobe radius for both the lambda and binding energy calculations, as well as for the calculation of the post-CEE separation.~
2. Use the stellar radius as an input (instead of Roche-lobe radius).

**Versioning (please complete the following information):**
 - OS: MacOS Catalina
 - COMPAS [e.g. v02.11.01]

**Additional context**
N/A

**Comment by avigna** [comment]
> All the BaseBinaryStar::ResolveCommonEnvelopeEvent function should only read with Rsol units, instead of a mixture between AU and Rsol.

This was sorted out as part of https://github.com/TeamCOMPAS/COMPAS/pull/303.

**Comment by avigna** [comment]
> Use the stellar radius as an input (instead of Roche-lobe radius).

This is already done as has been fixed during the rewriting of the code.

```
	m_Star1->CalculateLambdas(m_MassEnv1);                                                                              // calculate lambdas for star1
	m_Star2->CalculateLambdas(m_MassEnv2);                                                                              // calculate lambdas for star2

    m_Star1->CalculateBindingEnergies(m_Mass1Final, m_MassEnv1, m_Star1->Radius());                                     // calculate binding energies for star1 (uses lambdas)
    m_Star2->CalculateBindingEnergies(m_Mass2Final, m_MassEnv2, m_Star2->Radius());                                     // calculate binding energies for star2 (uses lambdas)

    m_Star1->CalculateCommonEnvelopeValues();                                                                           // calculate common envelope values for star1
    m_Star2->CalculateCommonEnvelopeValues();                                                                           // calculate common envelope values for star2
```

Both code refactoring and issue #303 sorted out this. No need to do anything else.

**Comment by avigna** [comment]
This actually might explain why we had an immediate drop in DNS rates in the refactored version with respect to Legacy.

**Comment by ilyamandel** [comment]
@avigna -- I see 
double k1            = m_Star1->IsOneOf(COMPACT_OBJECTS) ? 0.0 : (2.0 / (lambda1 * alphaCE)) * m_Star1->Mass() * m_MassEnv1 / rRLd1Rsol;
     double k2            = m_Star2->IsOneOf(COMPACT_OBJECTS) ? 0.0 : (2.0 / (lambda2 * alphaCE)) * m_Star2->Mass() * m_MassEnv2 / rRLd2;		     double k2            = m_Star2->IsOneOf(COMPACT_OBJECTS) ? 0.0 : (2.0 / (lambda2 * alphaCE)) * m_Star2->Mass() * m_MassEnv2 / rRLd2Rsol;
on lines 1849 and 1850 of BaseBinaryStar.cpp in #303  (see https://github.com/TeamCOMPAS/COMPAS/pull/303/files ).  That looks like RL radius, not stellar radius.

**Comment by ilyamandel** [comment]
Fixed by @avigna in #307 .
